.palantir/revapi.yml

```diff
@@ -66,6 +66,9 @@ acceptedBreaks:
       old: "method void org.apache.iceberg.io.DataWriter<T>::add(T)"
       justification: "Removing deprecated method"
   "1.1.0":
+    org.apache.iceberg:iceberg-api:
+    - code: "java.method.addedToInterface"
+      justification: "Add metricsreporter to Scan"
```

Comment:
Instead of modifying checks, we should throw `UnsupportedOperationException` in the default implementation. Just like we do in some methods in `TableScan` today.

Reply:
+1 to that